### PR TITLE
Added PGSQL Support

### PR DESCRIPTION
**Pull request by rwdim**
Added functional postgresql support to this already fantastic utility.

### CODE REVIEW

**Comment by CristianLlanos** [comment]
Thanks for your contribution. Would you mind updating your branch with the latest changes, please? I'll for @gareth-ib's feedback and after that I'll merge your PR.

File: src/Meta/SchemaManager.php [hunk 1]
```diff
@@ -44,23 +47,10 @@ class SchemaManager implements IteratorAggregate
...
-    /**
-     * Load all schemas from this connection.
-     */
-    public function boot()
-    {
         if (! $this->hasMapping()) {
             throw new RuntimeException("There is no Schema Mapper registered for [{$this->type()}] connection.");
         }
-
-        $schemas = forward_static_call([$this->getMapper(), 'schemas'], $this->connection);
```
Comment (CristianLlanos) [inline src/Meta/SchemaManager.php]: Why did you remove these lines?

Comment (CristianLlanos) [reply src/Meta/SchemaManager.php]: Please add them back. It's essential for the mapping process.

Comment (rwdim) [reply src/Meta/SchemaManager.php]: Except they render the code useless. When they are in, the code uses all schemas and generates all the tables from all the schemas, at least when using PostgreSQL. I tested it with and without on both and the functionality worked on both MySQL and PostgreSQL with the changes I made.
I'll do another round of testing, but I'm fairly certain I'll get the same result.

Comment (rwdim) [reply src/Meta/SchemaManager.php]: It's important to note that I am running the latest Laravel Nova, and the behavior I experienced is using that code base.

**Comment by rwdim** [comment]
I checked out the code, created an upstream reference to your repo, did a `git fetch upstream` and no changes were pulled as I had checked out your code the same day I made the changes.  I restored your original code in SchemaManager and re-tested using MySQL and PGsql and they both worked fine, so I must have had a configuration issue at the time I did the original testing.

File: src/Meta/SchemaManager.php (hunk 1 above)
Comment (CristianLlanos) [reply src/Meta/SchemaManager.php]: You wouldn't notice the difference unless PostgreSQL supports multiple database in a single database deamon as MySQL does. For instance, you can have a MySQL database `a` with a table `a_table` and a database `b` with a table `b_table` having a foreign key pointing to `a.a_table`. 😅 It's not easy to make a good example.

**Comment by rwdim** [comment]
Postgres absolutely supports multiple databases in a single daemon... which
is probably what I saw, but the issue probably arose because I was using
an account that had Superuser access.

I will test again with multiple databases, with and without Superuser
access.

Cheers
Randy

**Comment by rwdim** [comment]
The issue isn't about the same type of cross-database querying that you can do with MySql. That functionality isn't inherently available in PostgreSQL, at least not up to 10.0, without the use of plugins.

The issue is user rights and database isolation. PG allows each daemon access to all databases it knows about, and the only thing keeping an account from them is the role rights. In earlier versions, PG was less strict about enforcing those role rights.

In PG 8.3, which I still have a couple old dev databases running on, when you execute:

php artisan code:models  --schema=public
It spins forever, opens many, many connections, and generates all models for all databases your user has access to with public schemas.

When I do the same thing on 9.5, it isolates the user to the database and schema you've specified in your configuration.

Very odd behavior, but it works normally in current 9.5 and later versions.

**Comment by CristianLlanos** [comment]
Wow! What do you suggest, @rwdim? Should we do something about it?

**Comment by rwdim** [comment]
I've always stopped it 'cuz netstat shows 100+ connections... At any rate,
it only happens with PG8.x, so I wouldn't spend a moment fixing it. If I
was managing those databases properly (they're of little significance, so
upgrading them is not a priority), I'd have them on 10.x and would never
have seen the issue.

Bugs in prior major versions that can't be replicated in the current
version should be immediately closed, IMHO...

**Comment by CristianLlanos** [comment]
Thanks :+1:

**Comment by rwdim** [comment]
Found one issue... I neglected to change the default table names to ignore...
[Commit Link](https://github.com/reliese/laravel/commit/35fae994d98e2ff7ae3b8c56439abff4e4574d33)

Should I do a new PR?

Also started adding support for automatic Seeder and Migration support...

**Comment by CristianLlanos** [comment]
It'd be better to add those features in different PRs, unless this PR needs them.

**Comment by rwdim** [comment]
Agreed.. Ill add them later under a new pr

**Comment by krlicmuhamed** [comment]
Any chance for publishing this PGSQL support?

**Comment by timhaak** [comment]
> Any chance for publishing this PGSQL support?

As they don't seem to be accepting pull requests and maintaining. 

I've forked the repository and merged some of the pull requests.

https://github.com/haakco/reliese-laravel-models

https://packagist.org/packages/haakco/reliese-laravel-models

**Comment by CristianLlanos** [comment]
@rwdim, thanks for all the effort you put into it.

**Comment by rwdim** [comment]
nah.. been down this road before.. my code works beautifully and has since
I did my PR.

cheers
